core/src/components/button/button.tsx

```diff
@@ -182,57 +207,48 @@ export class Button implements ComponentInterface, AnchorInterface, ButtonInterf
         return el;
       }
     }
-    return null;
+    if (form !== undefined) {
+      /**
+       * The developer specified a form selector for
+       * the button to submit, but it was not found.
+       */
+      if (typeof form === 'string') {
+        printIonWarning(
```

Comment:
I don't believe this warning prints under the conditions it specifies. Let me know if I misread it, this is what I am observing:

If the `form` is a string, but the matched element is not an `HTMLFormElement`, this warning would log.

This statement would not return early, but would match the current conditionals for this warning: https://github.com/ionic-team/ionic-framework/blob/85058cc1cf9fab4ec979224cb6a5e1437a664b58/core/src/components/button/button.tsx#L204-L209

I think instead we want to check here: https://github.com/ionic-team/ionic-framework/blob/85058cc1cf9fab4ec979224cb6a5e1437a664b58/core/src/components/button/button.tsx#L205 if `el` is `null`/`undefined` log this warning. 

I think then the message for the `form` element being invalid can be the default behavior if neither of these conditions returns: https://github.com/ionic-team/ionic-framework/blob/85058cc1cf9fab4ec979224cb6a5e1437a664b58/core/src/components/button/button.tsx#L201-L209

Reply:
I updated the code and added comments.